flag_test.go

```diff
@@ -0,0 +1,40 @@
+package gosec_test
+
+import (
+	"flag"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/securego/gosec/v2/cmd/vflag"
+)
+
+var _ = Describe("Cli", func() {
+	Context("vflag test", func() {
+		It("value must be empty as parameter value contains invalid character", func() {
+			os.Args = []string{"gosec", "-test1=-incorrect"}
+			f := vflag.ValidatedFlag{Error: "Flag value cannot start with -."}
+			flag.Var(&f, "test1", "")
+			flag.CommandLine.Init("test1", flag.ContinueOnError)
+			flag.Parse()
```

Comment:
We should check if the `flag.Parse` returns an error.

Reply:
From the API the `flag.Parse()` does not return anything https://cs.opensource.google/go/go/+/refs/tags/go1.16:src/flag/flag.go;l=1020 or am I looking it the wrong one ?

Reply:
You are right. I looked at https://pkg.go.dev/flag#FlagSet.Parse.